scala/integration/maven/src/org/jetbrains/plugins/scala/project/maven/ScalaMavenImporter.scala

```diff
@@ -180,4 +189,12 @@ private object ScalaMavenImporter {
 
     def valid: Boolean = compilerPlugin.isDefined && compilerVersion.isDefined
   }
+
+  /**
+    * Represents a Maven artifact by group, artifact, version and optional classifier.
+    * Similar to [[org.jetbrains.idea.maven.model.MavenId]], but supports classifier.
```

Comment:
If you define the class as `MavenGAVC(id: MavenId, classifier: Option[String] = None)`, then this fact would be apparent, and the comment would be redundant, as well as the additional `mavenId` method.

Reply:
Thanks for the review @pavelfatin. 

I've added another commit which defines MavenGAVC in terms of MavenId, but I'm not sure it really improves things overall. I found that the function `localPathTo` which uses `MavenId`/`MavenGAVC` got pretty ugly (eg `id.mavenId.getArtifactId`) until I defined an `apply` method and direct accessors for `groupId`, `artifactId` and `version`.

My idea of `MavenGAVC` was really a complete replacement for `MavenId`, which surprised me by not supporting classifier. Honestly I don't think there is much to be gained by reusing a Java class (`MavenId`) which is a one-line case class in Scala. It doesn't really stand alone as a concept once you need classifier, which is why I ended up using MavenGAVC all over the file.

The branch is open for editing if I've missed your point entirely and you want to try something else. Or if you want to drop that last commit before merging. ;-)

Reply:
Hmm... with the accessors and the extractor it's indeed not much better :)

The main proposal was to explicitly use a classifier string, instead of the `MavenGVAC`, only where needed, because:
* The classifier is needed only in some places, yet the "Id" -> "GVAC" has to be update everywhere.
* The "GAVC" thing is not obvious.
* "GAVC" is still called "id" everywhere: `def resolve(id: MavenGAVC)`.
* The `MavenId` is accessed as `.mavenId`, even though the "GAVC" is already "Maven".

What if we use the original variant, but:
1. Import the IDEA's `MavenId` as `IdeaMavenId`.
2. Rename the `MavenGVAC` to `MavenId`.
3. Define `asIdea` method in our `MavenId` (by analogy with the `asScala`).

This will keep the familiar name and will minimize the unrelated changes.

Reply:
Hi @pavelfatin I've implemented your suggestions, but I think perhaps it would be better to omit `asIdea` (it throws away the classifier, so it's a bit of a leaky abstraction).

Alternatively, I suppose `asIdea` could return a tuple: `(MavenId, Option[Classifier])` to avoid the leak, but that tuple isn't really an IDEA concept, so I don't really favour this choice.

The `resolve` method which now calls `asIdea` could simply construct the `MavenArtifactInfo` using the group/artifact/version (and classifier) directly from the case class, in which case `asIdea` isn't needed.


I would normally try to avoid calling two different things by the same name (`MavenId`), but they are in different packages and this one's private, so it's just a matter of style. I don't suppose you like the name `MavenCoord` for the case class formerly known as `MavenGAVC`?

Reply:
> The resolve method which now calls `asIdea` could simply construct the `MavenArtifactInfo` using the group/artifact/version (and classifier) directly from the case class, in which case `asIdea` isn't needed.

True, that would be even better (because `asIdea` is indeed a leaky abstraction).

> I would normally try to avoid calling two different things by the same name (MavenId), but they are in different packages and this one's private, so it's just a matter of style. I don't suppose you like the name MavenCoord for the case class formerly known as MavenGAVC?

I also don't typically advocate reusing a name for two different things, but comparing to the "GAVC" (or "Coord", for that matter) abbreviation, reusing the name seems like a lesser evil :) BTW, it's not that uncommon - the Scala standard library re-uses the (publicly accessible) names of generic / mutable / immutable collections (though, that's why we have to add the `mutable.` prefix).

Feel free to add the `MavenArtifactInfo` creation, and we'll merge the changes.